plugins/catalog-backend/src/ingestion/LocationReaders.ts

```diff
@@ -258,24 +258,31 @@ export class LocationReaders implements LocationReader {
     }
   }
 
-  private async readLocation(
-    location: LocationSpec,
-  ): Promise<LocationProcessorResult> {
-    let locationResult: LocationProcessorResult | undefined;
+  private async readLocation(location: LocationSpec): Promise<Buffer> {
+    let data: Buffer | undefined = undefined;
+    let error: Error | undefined = undefined;
 
     await this.handleLocation(
       {
         type: 'location',
         location,
         optional: false,
       },
-      r => (locationResult = r),
+      output => {
+        if (output.type === 'error' && !error) {
+          error = output.error;
```

Comment:
Does it make sense to collect errors after the first or no?

Reply:
Or is it because we only `emit` one error anyway? Just wondered if theres going to be anything of value that might come in after the fact that `error` is set?

Reply:
It's because of the callback style. Can't know that we can throw in there and there's no way of telling the caller to stop processing

We could TRY to throw but it would be up to the implementation how it happens and that's leaky to make assumptions on

Anyway it's just an optimization if we stopped early, cba now

Reply:
No worries - was more of a question rather something to block this.

Reply:
haha, reading that again it sounded harsh, not meant like that at all :)